homeassistant/components/analytics/analytics.py

```diff
@@ -75,12 +79,116 @@
     ATTR_USER_COUNT,
     ATTR_UUID,
     ATTR_VERSION,
+    DOMAIN,
     LOGGER,
     PREFERENCE_SCHEMA,
     STORAGE_KEY,
     STORAGE_VERSION,
 )
 
+DATA_ANALYTICS_MODIFIERS = "analytics_modifiers"
+
+type AnalyticsModifier = Callable[
+    [HomeAssistant, AnalyticsInput], Awaitable[AnalyticsModifications]
+]
+
+
+@singleton(DATA_ANALYTICS_MODIFIERS)
+def _async_get_modifiers(
+    hass: HomeAssistant,
+) -> dict[str, AnalyticsModifier | None]:
+    """Return the analytics modifiers."""
+    return {}
+
+
+@dataclass
+class AnalyticsInput:
+    """Analytics input for a single integration.
+
+    This is sent to integrations that implement the platform.
+    """
+
+    device_ids: Iterable[str] = field(default_factory=list)
+    entity_ids: Iterable[str] = field(default_factory=list)
+
+
+@dataclass
+class AnalyticsModifications:
+    """Analytics config for a single integration.
+
+    This is used by integrations that implement the platform.
+    """
+
+    remove: bool = False
+    devices: Mapping[str, DeviceAnalyticsModifications] | None = None
+    entities: Mapping[str, EntityAnalyticsModifications] | None = None
+
+
+@dataclass
+class DeviceAnalyticsModifications:
+    """Analytics config for a single device.
+
+    This is used by integrations that implement the platform.
+    """
+
+    remove: bool = False
+
+
+@dataclass
+class EntityAnalyticsModifications:
+    """Analytics config for a single entity.
+
+    This is used by integrations that implement the platform.
+    """
+
+    remove: bool = False
+    capabilities: dict[str, Any] | None | UndefinedType = UNDEFINED
```

Comment:
So there are two options for this one naming wise.

* `capabilities` is the key in the entity registry where this information was sourced from
* `capability_attributes` is the entity property where the entity registry got the information from